packages/rsocket-core/src/RSocketResumableTransport.js

```diff
@@ -275,13 +275,18 @@ export default class RSocketResumableTransport implements DuplexConnection {
     if (this._isTerminated()) {
       return;
     }
-    if (error) {
-      this._setConnectionStatus({error, kind: 'ERROR'});
-    } else {
-      this._setConnectionStatus(CONNECTION_STATUS.CLOSED);
-    }
+
+    const status = error ? {error, kind: 'ERROR'} : CONNECTION_STATUS.CLOSED;
+    this._setConnectionStatus(status);
```

Comment:
A question I have with attaching the error to the emitted status is whether or not this would always be desirable. For instance, if the error which occurs during setup is the result of Authn or Authz, the server may not want to communicate the exact error which occurred down to the caller, as they may be untrusted.

It may be sufficient to state that implementers of an RSocket server via `rsocket-js` would be expected to massage their error has necessary in `getRequestHandler` before finally throwing to prevent leaking any sensitive information to the client, but would also be interested to hear anyone else's opinion on the matter.

Reply:
If the error has reached this code, hasn't it already been arrived at the client?

Reply:
Yep, you are entirely correct! Apologies for the confusion.